R/pkg/R/install.R

```diff
@@ -152,6 +152,9 @@ install.spark <- function(hadoopVersion = "2.7", mirrorUrl = NULL,
                      })
   if (!tarExists || overwrite || !success) {
     unlink(packageLocalPath)
+    if (success) {
```

Comment:
why should this be in inside this `if` block for `overwrite || !success` -- Can't we just have it outside this if as an else below ?

Reply:
basically if `success` is TRUE then only the other cases matter: `!tarExists || overwrite` - that's the exact condition where the download would have occurred in L126 which is the else case of `tarExists && !overwrite` on L123

Reply:
Hmm ok that still looks weird in the code. Maybe add a comment before this of the form `If we downloaded a tarfile or overwrote it, set sparkDownloaded flag` ?